### PR TITLE
Add Iris and Wine datasets

### PR DESCRIPTION
**Pull request by cocoa-xu**
Add Iris and Wine datasets. #16

### CODE REVIEW

File: lib/scidata/wine.ex [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    base_url = opts[:base_url] || @base_url
+    dataset_file = opts[:dataset_file] || @dataset_file
+
+    label_attr =
+      Utils.get!(base_url <> dataset_file, opts).body
+      |> String.split()
+      |> Enum.map(&String.split(&1, ","))
+      |> Enum.map(fn row ->
+          [label | val_list] = row
+          {label, ""} = Integer.parse(label)
```
Comment (t-rutten) [inline lib/scidata/wine.ex]:
```suggestion
          label = String.to_integer(label)
```

File: lib/scidata/wine.ex [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          val_list =
+            Enum.map(val_list, fn val ->
+              {val, ""} =
+                case val do
+                  "." <> _other ->
+                    Float.parse("0" <> val)
+
+                  _ ->
+                    Float.parse(val)
+                end
```
Comment (t-rutten) [inline lib/scidata/wine.ex]:
```suggestion
                Float.parse("0" <> val)
```
We can prepend the 0 to values that don't start with "." and they'll be parsed the same. Control flow is a bit simpler that way :)

File: lib/scidata/iris.ex [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            Enum.map([sl, sw, pl, pw], fn val ->
+              {val, ""} = Float.parse(val)
+              val
+            end)
+          label = get_label(label)
+          [label, sl, sw, pl, pw]
+      end)
+    labels = Enum.map(label_attr, &hd(&1))
+    attributes = Enum.map(label_attr, &tl(&1))
+    {attributes, labels}
```
Comment (t-rutten) [inline lib/scidata/iris.ex]:
```suggestion
     Utils.get!(base_url <> dataset_file, opts).body
    |> String.split()
    |> Enum.reverse()
    |> Enum.reduce({[], []}, fn row_str, {feature_acc, label_acc} ->
      row = String.split(row_str, ",")
      {features, [label]} = Enum.split(row, 4)

      features =
        Enum.map(features, fn val ->
          {val, ""} = Float.parse(val)
          val
        end)

      label = get_label(label)

      {[features | feature_acc], [label | label_acc]}
    end)
```
We can accumulate the labels and attributes in one pass :) We can apply this pattern to the wine dataset as well.

**Review by t-rutten — COMMENTED** [review]
@cocoa-xu you rock! Thanks for another great PR. Changes look great, I left some minor suggestions. Can you please rebase main and run `mix format` as well?

**Comment by cocoa-xu** [comment]
Thanks for the suggestions! I just committed them, hope I didn't miss anything! :)

**Review by t-rutten — APPROVED** [review]
Wonderful! 👏 Thanks for updating readme and documentation sections as well.

**Pull request merged**